CMakeLists.txt

```diff
@@ -459,6 +459,12 @@ try_append_cxx_flags("-fstack-reuse=none" TARGET core_interface)
 if(ENABLE_HARDENING)
   add_library(hardening_interface INTERFACE)
   target_link_libraries(core_interface INTERFACE hardening_interface)
+
+  target_compile_options(hardening_interface INTERFACE
+    -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_DEBUG
+    -D_GLIBCXX_ASSERTIONS
+  )
```

Comment:
Should this be set to `_LIBCPP_HARDENING_MODE_FAST` or `_LIBCPP_HARDENING_MODE_DEBUG` based on the build type?

Reply:
I think so, "yes". Done.